scripts/install/production/roles/tgt/tasks/main.yml

```diff
@@ -0,0 +1,104 @@
+---
+#Install tgt and its necessary dependencies
+
+- name: Install packages for tgt CentOS installation
+  package: name={{ item }} state=latest
+  with_items:
+   - gcc
+   - cpan
+   - make
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Change SELinux to permissive for CentOS
+  selinux:
+    policy: targeted
+    state: permissive
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Install EPEL repo for CentOS
+  yum:
+    name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+    state: present
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Install git
+  package:
+    name: git 
+    state: present
+  become: true
+
+- name: Install pip
+  package:
+    name: python-pip
+    state: present
+  become: true
+
+- name: Install TGT for Ubuntu
+  apt: name={{ item }} state=latest
+  with_items:
+   - tgt
+   - tgt-rbd
+  become: true
+  when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+- name: Install Perl General config for CentOS
+  yum: 
+    name: perl-Config-General.noarch
+    state: present
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Install headers for tgt source compilation for CentOS
+  yum: name={{ item }} state=latest
+  with_items:
+   - librbd1-devel
+   - librados2-devel
+   - libvirt
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Install tgt from source for CentOS
+  git: 
+    repo: https://github.com/fujita/tgt
+    dest: "{{playbook_dir}}/tgt"
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Make tgt for CentOS
+  command: "{{ item }}"
+  args:
+    chdir: "{{ playbook_dir }}/tgt"
+  with_items:
+    - "/usr/bin/make CEPH_RBD=1 clean"
+    - "/usr/bin/make CEPH_RBD=1"
+    - "/usr/bin/make CEPH_RBD=1 install"
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Copy tgtd.service to system folder for CentOS
+  copy:
+    src: "{{playbook_dir}}/tgt/scripts/tgtd.service"
+    dest: /usr/lib/systemd/system/
+  become: true
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+
+- name: Allow tcp via iptables for CentOS
+  iptables:
+    chain: INPUT
+    protocol: tcp
+    match: tcp
+    destination_port: 3260
+    jump: ACCEPT
```

Comment:
This rule wouldn't persist through a reboot. We should document this somewhere. For centos 7 and up, firewalld is what's used to manage iptables rules; there's no iptables service that can save the rules (it can be installed separately, but it didn't save the configuration for me :/) 

is there a way to setup these rules using firewalld here?

https://www.rootusers.com/how-to-open-a-port-in-centos-7-with-firewalld/

Reply:
I removed iptables and added firewalld